src/meta_schedule/space_generator/post_order_apply.cc

```diff
@@ -55,8 +55,10 @@ class BlockCollector : public tir::StmtVisitor {
     CHECK(block_names_.count(block->name_hint) == 0)
         << "Duplicated block name " << block->name_hint << " in function " << func_name_
         << " not supported!";
-    block_names_.insert(block->name_hint);
-    blocks_to_collect_.push_back(block->name_hint);
+    if (block->name_hint != "root") {
```

Comment:
It's a bit tricky. Can we skip all opaque blocks (i.e. blocks without block vars)?

Reply:
Thanks for the suggestion, after some more dig in I found there's no issue with the correct TIR so I would simply close the PR.